docs/usage.md

```diff
@@ -1,103 +1,480 @@
+    +------------------------+-------------------------+-----------------------------------------+----------+-------------+----------------------+----------------------------------------------------------------+--------+
+    |                        |                         |                                         |          |             |                      |                                                        Overall |   FAIL |
+    +------------------------+-------------------------+-----------------------------------------+----------+-------------+----------------------+----------------------------------------------------------------+--------+
+    ```
+
+=== "junit"
+
+    ```
+    <testsuites name="Regula">
+      <testsuite name="infra_cfn/invalid_long_description.yaml" tests="2">
+        <testcase name="InvalidManagedPolicy01" classname="AWS::IAM::ManagedPolicy" assertions="1">
+          <failure message="IAM policies must have a description of at least 25 characters" type="long_description_cfn">Rule ID: CUSTOM_0001&#xA;Rule Name: long_description_cfn&#xA;Severity: Low&#xA;Message: IAM policies must have a description of at least 25 characters</failure>
+        </testcase>
+        <testcase name="ValidManagedPolicy01" classname="AWS::IAM::ManagedPolicy" assertions="1"></testcase>
+      </testsuite>
+    </testsuites>
+    ```
+
+=== "tap"
+
+    ```
+    not ok 0 InvalidManagedPolicy01: IAM policies must have a description of at least 25 characters
+    ok 1 ValidManagedPolicy01: IAM policies must have a description of at least 25 characters
+    ```
+
+For more about Regula's output, see [Report Output](report.md).
+
+## repl
+
+```
+Start an interactive session for testing rules with Regula
+
+Usage:
+  regula repl [rego paths] [flags]
+
+Flags:
+  -h, --help        help for repl
+  -u, --user-only   Disable built-in rules
+```
+
+`regula repl` is the same as OPA's REPL ([`opa run`](https://www.openpolicyagent.org/docs/latest/#3-try-opa-run-interactive)), but with the Regula library and ruleset built in (unless you disable it with `--user-only`). Additionally, Regula's REPL allows you to generate [`mock_input`, `mock_resources`, and `mock_config`](development/rule-development.md#test-inputs) at runtime.
+
+To view this dynamically generated data, start by loading one or more IaC files or a directory containing IaC files. Note that Regula will only operate on individual files rather than interpreting the entire directory as a single configuration.
+
+### Examples
+This command loads the `infra` directory into the REPL:
+
+```
+regula repl ./infra
+```
+
+You'll see output like this:
+
+```
+Loaded 9 IaC configurations as test inputs
+Regula v0.8.0-dev - built with OPA v0.28.0
+Run 'help' to see a list of commands.
+>
+```
+
+Then you can [view the input](development/rule-development.md#viewing-test-inputs) using the format shown below:
```

Comment:
Some clarification we could put here or in the viewing-test-inputs section (if you think it's helpful) is that regula will automatically generate a package name for the input based on the file path. We need to abide by certain restrictions for package names, so we replace path separators with `.`s and most other characters (like `-`s) with `_`s.

I typically use the tab-completion in the repl to locate the test input after loading it, but maybe we need to do something more here - like print out the paths and packages that we loaded or provide some other way to easily get that information.

Reply:
Agreed -- another important clarification! And yeah, it would be really helpful to be able to see a list of the loaded package names. That would be a great enhancement.